### PR TITLE
Added Vault integration in oauth2-proxy image to retrieve secrets

### PR DESCRIPTION
**Pull request by iuriaranda**
Based the oauth2-proxy image on the official Vault docker image. The behavior can be tweaked via environment variables.

### CODE REVIEW

**Comment by SamClinckspoor** [comment]
can't you make it mandatory if vault auth is used to retrieve with a set role? instead of needing to provide all that info in `VAULT_AUTH_PARAM`

**Comment by iuriaranda** [comment]
@SamClinckspoor I think this allows you to be more flexible, as there can be other parameters that you need to provide, even with the `aws` auth backend, like `header_value`. I've based on how the vault-concourse integration works.

**Comment by SamClinckspoor** [comment]
isn't the header value the only part? I mean this is a very specific docker image.

Can you add an example in that case what the value should be?

**Comment by iuriaranda** [comment]
I think it's a rather generic image, it'll run the `oauth2_proxy` binary with whatever configuration you provide via command line parameters, and optionally it can be integrated with vault to fetch the client-id and client-secret for the proxy.

I'll add a usage example in the readme.

**Comment by iuriaranda** [comment]
@SamClinckspoor I've added an example of usage without Vault and another one with Vault.

**Review by SamClinckspoor — APPROVED** [review]
should `cookie-secret` also be in vault?

**Comment by iuriaranda** [comment]
@SamClinckspoor maybe, I'll first test the image with what we have and I'll make the appropriate modifications afterwards, as I think I'll have to change some other things as well.

File: oauth2-proxy/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+set -e
+
+if [ ! -z ${OAUTH_CREDS_VAULT_PATH} ]; then
+  if [ -z ${VAULT_URL} ] || [ -z ${VAULT_AUTH_PARAM} ] || [ -z ${VAULT_AUTH_METHOD} ]; then
```
Comment (MattiasGees) [inline oauth2-proxy/entrypoint.sh:6]: I don't think you need the extra check for `VAULT_AUTH_METHOD` as you already set it in your Dockerfile ;)

**Review by MattiasGees — APPROVED** [review]

File: oauth2-proxy/entrypoint.sh (hunk 1 above)
Comment (iuriaranda) [reply oauth2-proxy/entrypoint.sh:6]: Well, you could still override it to an empty value in docker run 😛

**Pull request merged**